app/scripts/offscreen.js

```diff
@@ -9,29 +10,51 @@ import { OffscreenCommunicationTarget } from '../../shared/constants/offscreen-c
  */
 export async function createOffscreen() {
   const { chrome } = globalThis;
-  if (!chrome.offscreen || (await chrome.offscreen.hasDocument())) {
+  if (!chrome.offscreen) {
     return;
   }
 
+  let offscreenDocumentLoadedListener;
   const loadPromise = new Promise((resolve) => {
-    const messageListener = (msg) => {
+    offscreenDocumentLoadedListener = (msg) => {
       if (
         msg.target === OffscreenCommunicationTarget.extensionMain &&
         msg.isBooted
       ) {
-        chrome.runtime.onMessage.removeListener(messageListener);
+        chrome.runtime.onMessage.removeListener(
+          offscreenDocumentLoadedListener,
+        );
         resolve();
       }
     };
-    chrome.runtime.onMessage.addListener(messageListener);
+    chrome.runtime.onMessage.addListener(offscreenDocumentLoadedListener);
   });
 
-  await chrome.offscreen.createDocument({
-    url: './offscreen.html',
-    reasons: ['IFRAME_SCRIPTING'],
-    justification:
-      'Used for Hardware Wallet and Snaps scripts to communicate with the extension.',
-  });
+  try {
+    await chrome.offscreen.createDocument({
+      url: './offscreen.html',
+      reasons: ['IFRAME_SCRIPTING'],
+      justification:
+        'Used for Hardware Wallet and Snaps scripts to communicate with the extension.',
+    });
+  } catch (error) {
+    if (offscreenDocumentLoadedListener) {
+      chrome.runtime.onMessage.removeListener(offscreenDocumentLoadedListener);
+    }
+    if (
+      error?.message?.startsWith(
+        'Only a single offscreen document may be created',
+      )
+    ) {
+      console.debug('Offscreen document already exists; skipping creation');
+    } else {
+      // Report unrecongized errors without halting wallet initialization
+      // Failures to create the offscreen document does not compromise wallet data integrity or
+      // core functionality, it's just needed for specific features.
+      captureException(error);
```

Comment:
Just updated this to catch all platform errors here (as suggested by @FrederikBolding on Slack) to avoid bricking the wallet if something else goes wrong here. We're still capturing the error in Sentry, and this is a non-essential step, so this should be a safe way to approach this for now.

ffb2c3a3689b20ee7f543dfb6be8658745297e8e